tiledb/sm/consolidator/fragment_consolidator.cc

```diff
@@ -446,16 +446,17 @@ void FragmentConsolidator::vacuum(const char* array_name) {
     array_dir.write_commit_ignore_file(commit_uris_to_ignore);
   }
 
-  // Delete the vacuum files.
   auto vfs = storage_manager_->vfs();
   auto compute_tp = storage_manager_->compute_tp();
-  vfs->remove_files(
-      compute_tp, filtered_fragment_uris.fragment_vac_uris_to_vacuum());
 
   // Delete fragment directories
   throw_if_not_ok(parallel_for(
       compute_tp, 0, fragment_uris_to_vacuum.size(), [&](size_t i) {
-        RETURN_NOT_OK(vfs->remove_dir(fragment_uris_to_vacuum[i]));
+        bool is_dir = false;
+        RETURN_NOT_OK(vfs->is_dir(fragment_uris_to_vacuum[i], &is_dir));
+        if (is_dir) {
+          RETURN_NOT_OK(vfs->remove_dir(fragment_uris_to_vacuum[i]));
+        }
```

Comment:
The first thing we should remove is the commits file to mark the fragment as no longer readable. One of the main issues is that if the remove_dir fails partially the commit file could still exist.

Reply:
@Shelnutt2 the fragment is marked no longer readable by the presence of its name in the .vac file, as long as the `.vac files` are deleted after succcessfully removing all fragment files and commit, I don't think the order between commit and frag dir matters.

Reply:
I disagree and think we must error on the side of caution and remove the commit file before we remove the directory.

Reply:
@robertbindar it should be an east change let’s get it done.

Reply:
done